### PR TITLE
[#4599][#4809] feat(core): Add tag support for column

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR proposes to add tag support for columns.

### Why are the changes needed?

With this PR, user could add and remove tags in the column level.

Fix: #4599 
Fix: #4809 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Add new UT and IT to test.

### CODE REVIEW

File: core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java [hunk 1]
```diff
@@ -69,101 +70,130 @@ public static long getMetadataObjectId(
...
+          break;
+
+        case CATALOG:
+          CatalogPO catalogPO = CatalogMetaService.getInstance().getCatalogPOById(objectId);
+          if (catalogPO != null) {
+            fullName =
+                fullName != null
+                    ? DOT_JOINER.join(catalogPO.getCatalogName(), fullName)
+                    : catalogPO.getCatalogName();
+            metadataType = null;
```
Comment (mchades) [inline core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java:115]: Don't change to `MetadataObject.Type.METALAKE` to obtain the metalake name?

Comment (jerryshao) [reply core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java:115]: No, type "CATALOG" is a top-level metadata object, we don't need to get metalake name.

File: docs/open-api/tags.yaml [hunk 2]
```diff
@@ -212,7 +212,7 @@ paths:
     post:
       tags:
         - tag
-      summary: Associate tags with metadata object
+      summary: Associate tags with metadata object, the supported metadata objects are CATALOG, SCHEMA, TABLE, FILESET, TOPIC, COLUMN
```
Comment (mchades) [inline docs/open-api/tags.yaml:215]: The summary is too long as a title (the web openAPI plugin will uses the summary as catalog title)

Comment (jerryshao) [reply docs/open-api/tags.yaml]: @mchades can you please check again?

**Comment by mchades** [comment]
Overall LGTM.

**Review by mchades — APPROVED** [review]
@yuqi1129 @jerqi Do you want to take another look?

**Pull request merged**